src/tags/capture_block.rs

```diff
@@ -40,25 +40,29 @@ impl Renderable for Capture {
 
 pub fn capture_block(
     _tag_name: &str,
-    arguments: &[Token],
-    tokens: &[Element],
+    mut arguments: TagTokenIter,
+    mut tokens: TagBlock,
     options: &LiquidOptions,
 ) -> Result<Box<Renderable>> {
-    let mut args = arguments.iter();
-    let id = match args.next() {
-        Some(&Token::Identifier(ref x)) => x.clone(),
-        x @ Some(_) | x @ None => return Err(unexpected_token_error("identifier", x)),
-    };
-
-    // there should be no trailing tokens after this
-    if let t @ Some(_) = args.next() {
-        return Err(unexpected_token_error("`%}`", t));
-    };
+    let id = arguments
+        .expect_next("Identifier expected")?
+        .expect_identifier()
+        .map_err(TagToken::raise_error)?
+        .to_string();
+
+    // no more arguments should be supplied, trying to supply them is an error
+    if let Some(token) = arguments.next() {
```

Comment:
Unsure if I missed this, is this in every tag/block?  If not, why is this different?

Reply:
I don't know if its on every tag/block, I think I just kept that in the ones that already did so in the old parser.

Though I totally think we should do something like that in every tag/block.